File: 2015/input_parsing/test_parse_input.py
import parse_input


def test_input_per_line_alpha():
    entries = parse_input.input_per_line("alpha_per_line_input.txt")
    assert entries[0] == "asdfasdf"
    assert entries[1] == "woeruwoer"
    assert entries[2] == "asdfasdf"
    assert entries[3] == "rufndnd dasdf"


def test_input_per_line_numeric():
    entries = parse_input.input_per_line("numeric_per_line_input.txt")
    assert entries[0] == "234234"
    assert entries[1] == "54565677"
    assert entries[2] == "23452345345456"
    assert entries[3] == "23423453454"
    assert entries[4] == "233232"


def test_input_per_line_alpha_numeric_symbols():
    entries = parse_input.input_per_line("alpha_numeric_symbols_per_line_input.txt")
    assert entries[0] == "dh#$liu\][asd"
    assert entries[1] == "lkhlh!@#*(&^@$%)<>?/\\"

File: input_readers/Ruby/test_parse_input.rb
require_relative "parse_input"
require 'test/unit'

class TestInputPerLine < Test::Unit::TestCase
    
    def test_alpha
        alpha_values = ["asdfasdf", "woeruwoer", "asdfasdf", "rufndnd dasdf"]
        assert_equal(alpha_values, input_per_line("alpha_per_line_input.txt"))
    end
end
